tailnet/controllers.go

```diff
@@ -104,6 +106,12 @@ type WorkspaceUpdatesController interface {
 	New(WorkspaceUpdatesClient) CloserWaiter
 }
 
+// DNSHostsSetter is something that you can set a mapping of DNS names to IPs on. It's the subset
+// of the tailnet.Conn that we use to configure DNS records.
+type DNSHostsSetter interface {
+	SetDNSHosts(hosts map[dnsname.FQDN][]netip.Addr) error
```

Comment:
Since the updater is maintaining it's own workspaces & agents state and just calling set each time, can we get rid of `(Conn).{Add|Remove}DNSHosts`? I don't see it being useful later

Reply:
Yeah, we could always add back in if there is a need.